src/snowflake/snowpark/modin/plugin/_internal/indexing_utils.py

```diff
@@ -911,21 +918,25 @@ def get_valid_col_positions_from_col_labels(
         indexer = []
         for col_loc_item in col_loc:
             # for each locator, we perform prefix matching for multiindex
-            try:
-                if is_scalar(col_loc_item):
-                    col_loc_item = (col_loc_item,)
-                item_indexer = columns.get_locs(col_loc_item)
-            except KeyError:
-                # Note Snowpark pandas will ignore out-of-bound labels
-                item_indexer = []
+            if is_scalar(col_loc_item):
+                col_loc_item = (col_loc_item,)
+            # May throw KeyError if any labels are not found in columns
```

Comment:
Is the KeyError not guaranteed to be thrown here?

Reply:
Only if labels are not found will the `KeyError` get trown